test/cluster/mzcompose.py

```diff
@@ -380,12 +380,14 @@ def find_command_history_metrics(c: Composition) -> Tuple[int, int, int, int]:
         controller_command_count < 100
     ), "controller history grew more than expected after peeks"
     assert (
-        controller_dataflow_count < 5
+        controller_dataflow_count == 1
     ), "more dataflows than expected in controller history"
     assert (
         replica_command_count < 100
     ), "replica history grew more than expected after peeks"
-    assert replica_dataflow_count < 5, "more dataflows than expected in replica history"
+    assert (
+        replica_dataflow_count == 1
```

Comment:
By the argument in https://github.com/MaterializeInc/materialize/pull/20251#issue-1781218423, it is not clear to me if a simple revert here is enough. For the argument of the PR, ISTM that it is important to ensure here that we do not observe zero. So it would be good to assert that we have at least 1 dataflow in the history (live or not), but we could have more and still be correct?

The test itself could potentially leave 2 dataflows behind after a reduction if it just happens that we sent a `CreateDataflows` and reduced the history before receiving the corresponding `Peek` and `AllowCompaction` for the slow-path `SELECT`. So to avoid a future flake, we might want to refine the condition here compared to what it was previously.

Reply:
You are right. I've adjusted the test now to check for >0 as well. It's not clear to me why we couldn't have more than 2 dataflows too, if compaction is sufficiently delayed, so I left the <5 as is.